source/mutex/src/raw_impls.rs

```diff
@@ -269,3 +269,79 @@ pub mod lock_api_0_4 {
         }
     }
 }
+
+#[cfg(feature = "std")]
+pub mod std {
+    //! TODO
+
+    use mutex_traits::RawMutex;
+    use core::{ptr::null_mut, sync::atomic::{AtomicPtr, Ordering}};
+    use std::sync::{Mutex, MutexGuard};
+
+    /// TODO
+    #[cfg_attr(feature = "fmt", derive(Debug))]
+    pub struct StdRawMutex {
+        // NOTE: We Box the mutex to ensure that even if we move `StdRawMutex` the
+        // MutexGuard (which points to the Mutex as a reference!) pointer is never
+        // invalidated
+        mtx: Box<Mutex<()>>,
+        guard: AtomicPtr<MutexGuard<'static, ()>>,
+    }
+
+    impl Drop for StdRawMutex {
+        fn drop(&mut self) {
+            // If we hold a guard, drop it. We can't race with anyone trying
+            // to unlock because we hold an exclusive mut ref
+            if self.is_locked() {
+                unsafe { self.unlock() }
+            }
+        }
+    }
+
+    impl StdRawMutex {
+        fn register_guard(&self, guard: MutexGuard<'_, ()>) {
+            let guard: &mut MutexGuard<'_, ()> = Box::leak(Box::new(guard));
+            let guard: *mut MutexGuard<'_, ()> = guard;
+            let guard: *mut MutexGuard<'static, ()> = guard.cast();
+            let old = self.guard.swap(guard, Ordering::AcqRel);
+            assert_eq!(old, null_mut());
+        }
+    }
+
+    unsafe impl RawMutex for StdRawMutex {
+        type GuardMarker = ();
+
+        #[inline]
+        #[track_caller]
+        fn lock(&self) {
+            let guard = self.mtx.lock().unwrap();
+            self.register_guard(guard);
+        }
+
+        #[inline]
+        #[track_caller]
+        fn try_lock(&self) -> bool {
+            match self.mtx.try_lock() {
+                Ok(g) => {
+                    self.register_guard(g);
+                    true
+                },
```

Comment:
`rustfmt` apparently wants you to remove the comma here: https://github.com/tosc-rs/scoped-mutex/actions/runs/10131659071/job/28014646552?pr=8#step:4:22
```suggestion
                }
```

this must be new, i've not seen it complain about that before...